class.jetpack.php

```diff
@@ -6929,22 +6930,22 @@ public static function build_raw_urls( $url ) {
 	 * Stores and prints out domains to prefetch for page speed optimization.
 	 *
 	 * @param mixed $new_urls
+	 * @param string | $relation_type | resource hint type
 	 */
-	public static function dns_prefetch( $new_urls = null ) {
+	public static function dns_prefetch( $new_urls = null, $relation_type = 'dns-prefetch' ) {
```

Comment:
You're right wrt having a `return` to solve part of the problem, but something I missed in the beginning. For the filter, it has two arguments -- urls, and relationship. Our first two args of `dns_prefetch` needs to match that. They use `prefetch` instead of `dns-prefetch`, so that's one thing.

Reply:
```suggestion
	public static function dns_prefetch( $urls = null, $relation_type = 'dns-prefetch' ) {
```